### PR TITLE
Add me.mitya57.ReText

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/retext-project/retext/issues/583
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build com.github.retext_project.retext

**Comment by Mailaender** [comment]
bot, build com.github.retext_project.retext

File: com.github.retext_project.retext.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  - name: python3-pyenchant
+    buildsystem: simple
+    sources:
+      - type: file
+        url: https://files.pythonhosted.org/packages/14/4b/6f7a3f2bb1e2fa4d3007126578cae0b9910ff46c4957bef5bd4b92733011/pyenchant-3.0.1.tar.gz
+        sha256: 1bd26a644abf80196a9de3f2d820ebafb7e7f78385e392ce77cb1552f164d559
+    build-commands:
+      - pip3 install --no-index --find-links="file://${PWD}" --prefix=${FLATPAK_DEST} pyenchant
+
+  - name: python3-docutils
```
Comment (TingPing) [inline com.github.retext_project.retext.yaml:63]: I highly recommend just running `[flatpak-pip-generator](https://github.com/flatpak/flatpak-builder-tools/blob/master/pip/flatpak-pip-generator)` and having it output a separate file. That allows updating modules to be a trivial single command. As-is you have to manually edit the main file.

Comment (Mailaender) [reply com.github.retext_project.retext.yaml]: That is indeed a lot more convenient. Thanks!

**Comment by Mailaender** [comment]
bot, build com.github.retext_project.retext

**Comment by Mailaender** [comment]
bot, build com.github.retext_project.retext

File: com.github.retext_project.retext.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+sdk: org.kde.Sdk
+base: com.riverbankcomputing.PyQt.BaseApp
+base-version: 5.15-21.08
+command: retext
+rename-desktop-file: me.mitya57.ReText.desktop
+rename-icon: retext
+rename-appdata-file: me.mitya57.ReText.appdata.xml
+finish-args:
+  - --share=ipc
+  - --socket=x11
```
Comment (barthalion) [inline com.github.retext_project.retext.yaml:13]:
```suggestion
  - --socket=fallback-x11
```

File: python3-Markdown.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (barthalion) [inline python3-Markdown.json:1]: What is it needed for?

**Comment by barthalion** [comment]
bot, build com.github.retext_project.retext

File: python3-Markdown.json (hunk 3 above)
Comment (Mailaender) [reply python3-Markdown.json:1]: Seems to be a bug in flatpak-pip-generator. Added it manually now.

**Comment by Mailaender** [comment]
bot, build com.github.retext_project.retext

**Comment by nedrichards** [comment]
why is the appropriate app-id not `me.mitya57.ReText`? That looks like a valid URL and has all the ReText content there as well. It's almost always a very bad idea to rename the app-id of something that already has one.

**Comment by Mailaender** [comment]
https://mitya57.me/ReText is 404 and I don't own the domain. The text at the top makes it look like that is a requirement.

**Comment by Mailaender** [comment]
bot, build me.mitya57.ReText

**Comment by hfiguiere** [comment]
> https://mitya57.me/ReText is 404 and I don't own the domain. The text at the top makes it look like that is a requirement.

the upstream developer seem to. It's only the domain (or subdomain / namespace) that is important.

image

**Comment by nedrichards** [comment]
Yeah, the text comes from the position of being the upstream developer or the company behind a project. Glad we could get that cleared up anyway

**Comment by nedrichards** [comment]
I get the following error when running

> libEGL warning: wayland-egl: could not open /dev/dri/renderD128 (No such file or directory)

which is because you need to add the --device=dri permission to allow Qt to use the GPU. Other than that, this looks good to me!

**Comment by Mailaender** [comment]
bot, build me.mitya57.ReText

**Pull request closed without merging**